File: builder-pattern/java/src/ManualBuilder.java
package src;

// 使用手册建造者类也实现了建造者接口
public class ManualBuilder implements Builder {
  private Manual manual;

  public void reset() {
    this.manual = new Manual();
  }

  public Builder setName(String name) {
    this.manual.setName(name);
    return this;
  }

  public Builder setScreen(Integer[] screen) {
    this.manual.setScreen(screen);
    return this;
  }

  public Builder setGPU(Integer no) {
    this.manual.setGpuType(no);
    return this;
  }

  public Manual getProduct() {
    return this.manual;
  }
}

File: builder-pattern/ts/src/ManualBuilder.ts
import { Builder } from './Builder.js'
import { Manual } from './Manual.js'

export class ManualBuilder implements Builder {
  manual: Manual | any
  constructor() {
    // this.reset()
  }

  reset() {
    this.manual = new Manual()
  }

  setName(name: string): ManualBuilder {
    this.manual.setName(name)
    return this
  }

  setScreen(screen: []): ManualBuilder {
    this.manual.setScreen(screen)
    return this
  }

  setGPU(no: number): ManualBuilder {
    this.manual.setGpuType(no)
    return this
  }

  getProduct() {
    return this.manual
  }
}
